holoviews/element/raster.py

```diff
@@ -692,7 +692,9 @@ def __init__(self, data, kdims=None, vdims=None, **params):
         if ((hasattr(data, 'shape') and data.shape[-1] == 4 and len(vdims) == 3) or
             (isinstance(data, tuple) and isinstance(data[-1], np.ndarray) and data[-1].ndim == 3
              and data[-1].shape[-1] == 4 and len(vdims) == 3) or
-            (isinstance(data, dict) and (*map(dimension_name, vdims), alpha.name) in data)):
+            (isinstance(data, dict) and (*map(dimension_name, vdims), alpha.name) in data) or
+            str(alpha) in getattr(data, "data_vars", [])
+        ):
```

Comment:
Not that it wasn't already but that's getting quite unwieldy.

Reply:
Agree.

Reply:
In that case please refactor 🙂

Reply:
I'm not sure what you mean. Do you want it to be put into a method?

Reply:
Refactored in https://github.com/holoviz/holoviews/pull/6422/commits/8f4378efad4e0298698345871c70f4ac720ba1b2

Which is what I think you mean